src/vm/eventpipebuffer.cpp

```diff
@@ -54,35 +54,65 @@ bool EventPipeBuffer::WriteEvent(Thread *pThread, EventPipeSession &session, Eve
         NOTHROW;
         GC_NOTRIGGER;
         MODE_ANY;
-        PRECONDITION(pThread != NULL);
         PRECONDITION(((size_t)m_pCurrent % AlignmentSize) == 0);
     }
     CONTRACTL_END;
 
     // Calculate the size of the event.
     unsigned int eventSize = sizeof(EventPipeEventInstance) + payload.GetSize();
+    DWORD osThreadId;
 
     // Make sure we have enough space to write the event.
     if(m_pCurrent + eventSize >= m_pLimit)
     {
         return false;
     }
 
+    if (pThread == NULL)
+    {
+        // For ThreadID just get the Current Thread ID
+        osThreadId = ::GetCurrentThreadId();
+    }
+    else
+    {
+        osThreadId = pThread->GetOSThreadId();
+    }
+
     // Calculate the location of the data payload.
     BYTE *pDataDest = m_pCurrent + sizeof(EventPipeEventInstance);
+    EventPipeEventInstance *pInstance;
 
     bool success = true;
     EX_TRY
     {
+        LPCGUID pThreadActivityID = pActivityId;
+
         // Placement-new the EventPipeEventInstance.
-        EventPipeEventInstance *pInstance = new (m_pCurrent) EventPipeEventInstance(
-            session,
-            event,
-            pThread->GetOSThreadId(),
-            pDataDest,
-            payload.GetSize(),
-            pActivityId,
-            pRelatedActivityId);
+
+        if (pThread == NULL)
```

Comment:
This block of code if very similar. Would it make more sense to write it this way:
```cpp
             pInstance = new (m_pCurrent) EventPipeEventInstance( 
                session, 
                event, 
                osThreadId, 
                pDataDest, 
                payload.GetSize(), 
                pThread == nullptr ? nullptr : pActivityId,
                pRelatedActivityId); 
```

Reply:
Yup. Vance commented on it above so I'm fixing it :)